src/ebay-accordion/accordion.tsx

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames'
+import React, { ComponentProps, FC, ReactElement, useRef } from 'react'
+import { EbayDetails, EbayDetailsProps } from '../ebay-details'
+import { filterByType } from '../utils'
+import { EbayEventHandler } from '../events'
+
+type ToggleProps = {
+    open: boolean
+    index: number
+}
+
+export type EbayAccordionProps = ComponentProps<'ul'> & {
+    size?: 'regular' | 'large'
+    autoCollapse?: boolean
+    children: ReactElement<EbayDetailsProps>[]
+    onToggle?: EbayEventHandler<HTMLDetailsElement, ToggleProps>
+}
+
+const EbayAccordion: FC<EbayAccordionProps> = ({
+    size,
+    autoCollapse,
+    className,
+    children,
+    onToggle,
+    'aria-roledescription': ariaRoledescription = 'accordion',
+    ...rest
+}: EbayAccordionProps) => {
+    const listRef = useRef<HTMLUListElement>(null)
+
+    const details = filterByType(children, EbayDetails)
+    if (!details.length) {
+        throw new Error(`EbayAccordion: Please use a
+        EbayDetails that defines the sections of the accordion`)
+    }
+
+    const detailsWithAccordionProps = details.map((child, detailsIndex) =>
+        React.cloneElement(child as ReactElement<EbayDetailsProps>, {
+            onToggle: (event, { open }) => {
+                if (autoCollapse && open) {
+                    const detailsElements = listRef.current?.querySelectorAll('details')
+                    detailsElements?.forEach((detail, elementIndex) => {
+                        if (elementIndex !== detailsIndex) {
+                            detail.open = false
+                        }
+                    })
+                }
+
+                onToggle?.(event, {
+                    open,
+                    index: detailsIndex
```

Comment:
@agliga @saiponnada, I added the `index` so the app have some more context. Let me know what you think